### PR TITLE
Improved performance of reading

### PR DESCRIPTION
**Pull request by till0xff**

### CODE REVIEW

File: Xcode/Sources/Socket.swift [hunk 1]
```diff
@@ -149,13 +149,9 @@ open class Socket: Hashable, Equatable {
     /// - Returns: A buffer containing the bytes read
     /// - Throws: SocketError.recvFailed if unable to read bytes from the socket
     open func read(length: Int) throws -> [UInt8] {
-        var buffer = UnsafeMutableBufferPointer<UInt8>.allocate(capacity: length)
-
-        let bytesRead = try read(into: &buffer, length: length)
-
-        let rv = [UInt8](buffer[0..<bytesRead])
```
Comment (till0xff) [inline Xcode/Sources/Socket.swift:156]: This line could run up to 25 seconds for `length` = 6 mb

**Comment by michaelenger** [comment]
@till0xff Could you update the [CHANGELOG.md](https://github.com/httpswift/swifter/blob/stable/CHANGELOG.md) file.

**Comment by till0xff** [comment]
@michaelenger  done

**Review by michaelenger — APPROVED** [review]
This looks good to me. What do you think @Vkt0r?

**Comment by Vkt0r** [comment]
Sorry for the delays guys, merging this. It will come in the next release for sure. Thanks for this @till0xff

**Pull request merged**